app/javascript/components/admin/AdminTable.js

```diff
@@ -335,6 +335,17 @@ class AdminTable extends React.Component {
     this.axiosAdminPostRequest(path, dataToSend, handleSuccess, handleError);
   };
 
+  /**
+   * Called when table is to be updated because of a search or sorting change
+   * @param {Object} query - sorting components of query from custom table
+   */
+  handleTableUpdate = query => {
+    const previous = this.state.query;
```

Comment:
When you want to use previous state before calling setState, you should use the optional `state` parameter in the setState function because it is asynchronous and you want to make sure you're using the right version at that point.

Reply:
addressed in https://github.com/SaraAlert/SaraAlert/pull/372/commits/d640c1359d82c869f49133e4d65b291185d55564

thanks @kylietmo !